src/amo/components/GetFirefoxButton/index.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+import * as React from 'react';
+import { compose } from 'redux';
+
+import { makeQueryStringWithUTM } from 'amo/utils';
+import translate from 'core/i18n/translate';
+import Button from 'ui/components/Button';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+export type Props = {|
+  addon: AddonType,
+|};
+
+type InternalProps = {|
+  ...Props,
+  i18n: I18nType,
+|};
+
+export const GetFirefoxButtonBase = (props: InternalProps) => {
+  const { addon, i18n } = props;
+
+  return (
+    <Button
+      buttonType="confirm"
+      href={`https://www.mozilla.org/firefox/new/${makeQueryStringWithUTM({
+        utm_content: addon.guid,
+      })}`}
+      puffy
+      className="GetFirefoxButton"
+    >
+      {i18n.gettext('Only with Firefox—Get Firefox Now')}
```

Comment:
given our discussion today, this should also be something we should be able to change

Reply:
I think you can leverage the `children` prop for that:

```js
<GetFirefoxButton>
  {i18n.gettext('Only with FF')}
</GetFirefoxButton>
```

Reply:
We could do that, but see my comment at https://github.com/mozilla/addons-frontend/pull/6925#discussion_r233590087. I think we may still have standard text for the button for the different types/styles, so rather than passing in the button text as children, maybe we should have `GetFirefoxButton` configure it based on type/style (which have yet to be defined)? I imagine there could be other aspects that need to be configured for different types (e.g., `puffy`) which is why I think this makes sense.

What do you think @willdurand / @rebmullin?

Reply:
That's a good idea.